.github/actions/setup-and-cache-rust/action.yml

```diff
@@ -19,6 +19,11 @@ inputs:
 runs:
   using: composite
   steps:
+    - name: no brew auto update
+      shell: bash
+      run: |
+        echo HOMEBREW_NO_AUTO_UPDATE=1 > $GITHUB_ENV
+        echo HOMEBREW_NO_INSTALL_UPGRADE=1 > $GITHUB_ENV
```

Comment:
Why not put this in the toplevel `env` block of each workflow? That way it would be always there.